src/app/(modals)/support/support-history.tsx

```diff
@@ -0,0 +1,196 @@
+import useUser from "@/hooks/useUser";
+import { normalizeStatus } from "@/services/normalizeStatus";
+import type { TicketDoc } from "@/types/TicketDoc";
+import { TicketStatus } from "@/types/TicketStatus";
+import {
+  collection,
+  FirebaseFirestoreTypes,
+  getFirestore,
+  onSnapshot,
+  orderBy,
+  query,
+  where,
+} from "@react-native-firebase/firestore";
+import dayjs from "dayjs";
+import { router, Stack } from "expo-router";
+import React, { useEffect, useState, type JSX } from "react";
+import {
+  ActivityIndicator,
+  FlatList,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+} from "react-native";
+
+export default function SupportHistory(): JSX.Element {
+  const user = useUser();
+  const [items, setItems] = useState<TicketDoc[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (!user?.uid) {
+      setItems([]);
+      setLoading(false);
+      return;
+    }
+
+    const q = query(
+      collection(getFirestore(), "support_tickets"),
+      where("uid", "==", user.uid),
+      orderBy("createdAt", "desc"),
+    );
+
+    const unsub = onSnapshot(
+      q,
+      (snap) => {
+        const arr: TicketDoc[] = snap.docs.map(
+          (d: FirebaseFirestoreTypes.QueryDocumentSnapshot) => {
+            const data = d.data() as Record<string, unknown>;
+            return {
+              id: d.id,
+              createdAt:
+                (data["createdAt"] as { toDate?: () => Date } | null) ?? null,
+              status:
+                (data["status"] as string | undefined) ?? TicketStatus.Open,
+              title: (data["title"] as string | undefined) ?? "",
+              message: (data["message"] as string | undefined) ?? "",
+            };
+          },
+        );
```

Comment:
```suggestion
        const arr: TicketDoc[] = snap.docs.map(
          (d: FirebaseFirestoreTypes.QueryDocumentSnapshot) => {
            const data = d.data() as TicketDoc;
            data.id = d.id;
            return data;
          },
        );
```

use same method in `support-detail.tsx` to convert type